electrum/lnchannel.py

```diff
@@ -1350,42 +1350,75 @@ def receive_revocation(self, revocation: RevokeAndAck):
                     error_bytes, failure_message = None, None
                 self.lnworker.htlc_failed(self, htlc.payment_hash, htlc.htlc_id, error_bytes, failure_message)
 
-    def extract_preimage_from_htlc_txin(self, txin: TxInput) -> None:
+
+    def extract_preimage_from_htlc_txin(self, txin: TxInput, is_confirmed: bool) -> None:
```

Comment:
can we have a more descriptive name than `is_confirmed`? e.g. `is_deeply_mined` or `is_deep`?
also, make it kwarg-only
```suggestion
    def extract_preimage_from_htlc_txin(self, txin: TxInput, *, is_deeply_mined: bool) -> None:
```

Reply:
done